editor/src/core/layout/layout-utils.ts

```diff
@@ -122,13 +129,39 @@ export function maybeSwitchLayoutProps(
   const newParent = MetadataUtils.findElementByElementPath(currentContextMetadata, newParentPath)
 
   let wasFlexContainer = MetadataUtils.isFlexLayoutedContainer(originalParent)
+  const oldParentMainAxis: 'horizontal' | 'vertical' | null = eitherToMaybe(
+    flatMapEither<string, JSXElementChild, 'horizontal' | 'vertical'>((parentElement) => {
+      if (isJSXElement(parentElement)) {
+        return FlexLayoutHelpers.getMainAxis(right(parentElement.props))
+      } else {
+        return leftEither('parent is not JSXElement')
+      }
+    }, newParent?.element ?? leftEither('no parent provided')),
+  )
+
   let isFlexContainer =
     parentLayoutSystem === 'flex' || MetadataUtils.isFlexLayoutedContainer(newParent)
+
+  const parentMainAxis: 'horizontal' | 'vertical' | null =
+    newParentMainAxis ??
+    // if no newParentMainAxis is provided, let's try to find one
+    // TODO maybe overkill
+    eitherToMaybe(
```

Comment:
This whole blocks _appears_ to be the same as the one above.

Reply:
fixed